api/java/java_model.go

```diff
@@ -51,6 +51,51 @@ func RequestToArtifact(vars map[string]string) (ja Artifact) {
 	}
 }
 
+// StorageIdentifierToArtifact : Convert a storage identifier into an Artifact.
+func StorageIdentifierToArtifact(id storage.Identifier) (ja Artifact) {
+
+	// Remove the initial slash if there is one (by mistake)
+	sanitized := strings.TrimPrefix(id.Key, id.Separator)
+
+	// Split the path into a set of
+	a := strings.Split(sanitized, id.Separator)
+
+	// General GAV parameters
+	filename, a := a[len(a)-1], a[:len(a)-1]
+	version, a := a[len(a)-1], a[:len(a)-1]
+	artifact, a := a[len(a)-1], a[:len(a)-1]
+	group := strings.Join(a, ".")
+
+	// Attempted this using string manipulation but couldn't - got scuppered
+	// by some weird version numbers (like 1 which got in the way of sha1)
+	// This seems a more robust way.
+	typeVar := ""
+	checksum := ""
+
+	idents := NewJavaFileList().FileTypes
+
+	for filetype := range idents {
+		if strings.HasSuffix(filename, filetype) {
+			checksumSlice := strings.Split(filetype, ".")
+			typeVar = checksumSlice[0]
+			if len(checksumSlice) > 1 {
+				typeVar = typeVar + "." + checksumSlice[1]
+				checksum = checksumSlice[1]
+			}
+			break
+		}
+	}
+
+	return Artifact{
+		Group:    group,
+		Artifact: artifact,
+		Version:  version,
+		Filename: filename,
+		Type:     typeVar,
+		Checksum: checksum,
```

Comment:
Is this not more accurately the checksum algorithm rather than the checksum itself?

Reply:
Agreed.